app/controllers/foreman_register/hosts_controller.rb

```diff
@@ -19,11 +19,20 @@ class HostsController < ::ForemanRegister::ApplicationController
     before_action :set_admin_user
     before_action :skip_secure_headers
     before_action :find_host
+    before_action :find_template
 
     def register
-      @host.setBuild
-      template = rendered_registration_template
-      render plain: template if template
+      if @template
+        @host.setBuild(validate_on_save: @host.managed?)
```

Comment:
why don't we rather disable the validation at https://github.com/theforeman/foreman/blob/2801602393aee2245a2f5066d3dc4e731fea40e2/app/models/host/base.rb#L42

it seems this patch is saying that it's a valid thing to set the build status for unmanaged host. In the RFC thread, I haven't seen any concerns with it, besides suggestion of getting rid of unmanaged host (which is a long term goal). We don't need this argument then at all.